packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1185,7 +1188,8 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
 
       renderObject.debugChildIntegrityEnabled = false; // Moving children will temporary violate the integrity.
       newChildren.keys.forEach(processElement);
-      if (_didUnderflow) {
+      // We don't worry about underflow if any child has been updated.
```

Comment:
This comment should include an explanation _why_ the underflow doesn't matter if children get updated.